os/board/rtl8720e/src/component/soc/amebalite/fwlib/ram_km4/ameba_ota_ram.c

```diff
@@ -480,5 +480,41 @@ int OTA_Image2SignatureCheck(uint32_t input_addr)
 	return TRUE;
 }
 
+/* Secure Key Derive Function structure */
+typedef struct {
+	unsigned char *secret;
+	size_t slen;
+	char *label;
+	unsigned char *context;
+	size_t clen;
+	unsigned char *dstbuf;
+	size_t dlen;
+} secure_kdf_struc;
+
+int Ameba_KeyDeriveFunc(const unsigned char *secret, size_t slen,
```

Comment:
All other security APIs are implemented in security_ameba_wrapper_tz.c or rtl_security_api_wrapper_tz.c files. Do you want to do only this here? Is there any difference between them?

Reply:
security_ameba_wrapper_tz.c is for SE-related APIs. This API is independent of SE.
It can work on its own without SE init.